### ISSUE TITLE
Type tests are failing on all PRs and preventing release

### ISSUE THREAD

**Issue by jni** [napari/napari#4470]
## 🐛 Bug

Here is the error in the release action:

https://github.com/napari/napari/runs/6286397863?check_suite_focus=true#step:6:45

```
make typestubs
make[1]: Entering directory '/home/runner/work/napari/napari'
python -m napari.utils.stubgen
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/runner/work/napari/napari/napari/utils/stubgen.py", line 205, in <module>
    generate_module_stub(mod)
  File "/home/runner/work/napari/napari/napari/utils/stubgen.py", line 178, in generate_module_stub
    _imports, stub = generate_class_stubs(obj)
  File "/home/runner/work/napari/napari/napari/utils/stubgen.py", line 133, in generate_class_stubs
    _imports, stub = generate_function_stub(method)
  File "/home/runner/work/napari/napari/napari/utils/stubgen.py", line 93, in generate_function_stub
    hints = get_type_hints(func, globalns)
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/site-packages/typing_extensions.py", line 876, in get_type_hints
    hint = typing.get_type_hints(
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/typing.py", line 1497, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/typing.py", line 292, in _eval_type
    return t._evaluate(globalns, localns, recursive_guard)
  File "/opt/hostedtoolcache/Python/3.9.12/x64/lib/python3.9/typing.py", line 554, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
Writing stub: /home/runner/work/napari/napari/napari/view_layers.pyi
TypeError: unsupported operand type(s) for |: 'type' and 'type'
make[1]: *** [Makefile:12: typestubs] Error 1
make: *** [Makefile:22: dist] Error 2
make[1]: Leaving directory '/home/runner/work/napari/napari'
```

However in PRs/main the error is different:

https://github.com/napari/napari/runs/6287580684?check_suite_focus=true#step:5:8

```
napari/plugins/_npe2.py:140: error: "Callable[[bool], None]" has no attribute "connect"  [attr-defined]
Found 1 error in 1 file (checked 29 source files)
make: *** [Makefile:18: typecheck] Error 1
```

Which weirdly is in code that hasn't changed recently.

I don't know whether the two errors are related but they are blocking release. @tlambert03 do you think you could have a look?

**Comment by tlambert03** [comment]
Probably due to recent mypy update.  Will have a look

**Comment by Czaki** [comment]
temporary downgrade mypy and fix after rc0 is released?

**Comment by tlambert03** [comment]
Sure that's fine too

**Comment by cnstt** [comment]
The "type tests" are still failing is this normal or has the 'fix' not fixed the issue?

**Comment by tlambert03** [comment]
you need to merge the main branch with the fix into your pr

**Comment by jni** [comment]
Fixed by #4475 and #4476. With thanks to @tlambert03!
